source/simulator/core.cc

```diff
@@ -1011,8 +1043,7 @@ namespace aspect
         if (parameters.use_discontinuous_temperature_discretization)
           face_coupling[x.temperature][x.temperature] = DoFTools::always;
 
-        // Only allocate composition 0 matrix if needed. Same as the non-DG case (see above)
-        if (parameters.use_discontinuous_composition_discretization && have_fem_compositional_field)
+        if (parameters.use_discontinuous_composition_discretization)
```

Comment:
does this work correctly with your change here? What if you have 0 compositional fields?